### PR TITLE
Add trait for easy registration with the World

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

**Comment by WaDelma** [comment]
So this is basically simpler version of #155? Should we just resurrect that one or simplify it to something like this?

**Comment by torkleyy** [comment]
The main motivation of @Aceeri's version was serialization and deserialization AFAIK (which has been solved by @omni-viral ), so I think a simpler solution is both sufficient and more practical.

File: src/common.rs [hunk 1]
```diff
@@ -150,6 +150,11 @@ impl Errors {
     }
 }
 
+/// Trait used to bundle up resources/components for easy registration with the `World`.
+pub trait WorldResourceRegistrator {
```
Comment (torkleyy) [inline src/common.rs]: I got a better name, today ;) You're already using "bundle" in the documentation, maybe that'd justify calling the trait `Bundle`?

File: src/common.rs [hunk 2]
```diff
@@ -150,6 +150,11 @@ impl Errors {
...
+    fn register_with_world(self, world: &mut World);
```
Comment (torkleyy) [inline src/common.rs]: Register implies a restriction to components, could you rename it to `add*`, please?

File: src/common.rs [hunk 3]
```diff
@@ -337,4 +342,24 @@ mod test {
...
+
+        pub struct TestRegistrator;
+
+        impl WorldResourceRegistrator for TestRegistrator {
+            fn register_with_world(self, world: &mut World) {
+                world.add_resource(SomeResource { v : 12 });
+            }
+        }
+
+        TestRegistrator.register_with_world(&mut world);
```
Comment (torkleyy) [inline src/common.rs]: We could add an impl for `World` here which adds an `add_bundle` method. Do you think that's useful @Rhuagh @WaDelma ?

File: src/world/mod.rs [hunk 4]
```diff
@@ -529,3 +534,8 @@ impl Default for World {
         }
     }
 }
+
+/// Trait used to bundle up resources/components for easy registration with `World`.
+pub trait Bundle {
+    fn add_to_world(self, world: &mut World);
```
Comment (torkleyy) [inline src/world/mod.rs]: Missing docs

Comment (Rhuagh) [reply src/world/mod.rs]: Yeah, I didn't know what to write :/

File: src/world/tests.rs [hunk 5]
```diff
@@ -101,3 +101,23 @@ fn delete_twice() {
...
+
+    impl Bundle for TestBundle {
+        fn add_to_world(self, world: &mut World) {
+            world.add_resource(SomeResource { v : 12 });
+        }
+    }
+
+    world.add_bundle(TestBundle);
+    assert_eq!(12, world.read_resource::<SomeResource>().v);
+}
```
Comment (Aceeri) [inline src/world/tests.rs]: New line

Comment (Rhuagh) [reply src/world/tests.rs]: It's missing .. fixed now.

**Comment by Xaeroxe** [comment]
What's blocking merging here?  This looks ready to go.

**Comment by torkleyy** [comment]
@kvark is on vacation and wants to approve it before merging, otherwise it's ready. Besides, we can't release anyways since there are still soundness issues with saveload stuff.

**Pull request merged**